### ISSUE TITLE
TreeAdmin issue: Cannot resolve keyword 'position' into field

### ISSUE THREAD

**Issue by kronok** [matthiask/feincms3#1]
Hey, I know this is super early stuff, but I ran into an issue trying to go from MPTT to cte-forest.

When I attempt to view the admin page for my model, I get:
```
Exception Value:

Cannot resolve keyword 'position' into field. Choices are: children, content_object, content_type, content_type_id, description, funnels, id, is_active, object_id, parent, parent_id, subscribers
```

Model:
```
from cte_forest.models import CTENode


class Step(CTENode):
    description = models.CharField(max_length=75, null=True, blank=True)
    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE, limit_choices_to=step_choices)
    object_id = GfkLookupField('content_type')
    content_object = GenericForeignKey('content_type', 'object_id')
    is_active = models.BooleanField(verbose_name="Active", default=True)
```

Admin:
```
from feincms3.admin import TreeAdmin


class StepAdmin(TreeAdmin):
    model = Step
    generic_raw_id_fields = ['content_object']
    raw_id_fields = ('parent',)
    list_display = ('indented_title', 'move_column')
```
Fresh app with fresh migrations.

I can do some tree related things:
```
>>> Step.objects.all()[0].as_tree()
{'path': ['1'], 'leaf': True, 'children': [], 'node': <Step: test>, 'branch': False, 'depth': 1, 'ordering': [1]}
```
 ...so the model seems to be working fine, it's just that admin view.

Here's the full traceback:
```
Traceback:

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  149.                     response = self.process_exception_by_middleware(e, request)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  147.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/lib/python3.4/contextlib.py" in inner
  30.                 return func(*args, **kwds)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/contrib/admin/options.py" in wrapper
  541.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/contrib/admin/sites.py" in inner
  244.             return view(request, *args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/contrib/admin/options.py" in changelist_view
  1577.             selection_note=_('0 of %(cnt)s selected') % {'cnt': len(cl.result_list)},

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/query.py" in __len__
  240.         self._fetch_all()

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/query.py" in _fetch_all
  1074.             self._result_cache = list(self.iterator())

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/query.py" in __iter__
  52.         results = compiler.execute_sql()

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in execute_sql
  837.             sql, params = self.as_sql()

File "/venv/smoothierecipes/lib/python3.4/site-packages/cte_forest/query.py" in as_sql
  411.         return CTECompiler.generate_sql(self.connection, self.query, _as_sql)

File "/venv/smoothierecipes/lib/python3.4/site-packages/cte_forest/query.py" in generate_sql
  332.         sql = as_sql()

File "/venv/smoothierecipes/lib/python3.4/site-packages/cte_forest/query.py" in _as_sql
  410.             return super(CTEQueryCompiler, self).as_sql(*args, **kwargs)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in as_sql
  382.             extra_select, order_by, group_by = self.pre_sql_setup()

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in pre_sql_setup
  49.         order_by = self.get_order_by()

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in get_order_by
  295.                     field, self.query.get_meta(), default_order=asc))

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in find_ordering_name
  566.         field, targets, alias, joins, path, opts = self._setup_joins(pieces, opts, alias)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in _setup_joins
  599.             pieces, opts, alias)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/query.py" in setup_joins
  1405.             names, opts, allow_many, fail_on_missing=True)

File "/venv/smoothierecipes/lib/python3.4/site-packages/django/db/models/sql/query.py" in names_to_path
  1330.                                      "Choices are: %s" % (name, ", ".join(available)))

Exception Type: FieldError at /admin/squeezemail/step/
Exception Value: Cannot resolve keyword 'position' into field. Choices are: children, content_object, content_type, content_type_id, description, funnels, id, is_active, object_id, parent, parent_id, subscribers
```

Any insight into what you think may be going on would be much appreciated.

**Comment by matthiask** [comment]
Thanks for the question! Not too early at all! feincms3 already runs on 3-4 sites in production.

I use a `position` field to order nodes within a tree level:

https://github.com/matthiask/feincms3/blob/master/feincms3/pages.py#L43

The field is defined on line 48 (right now).

How do you order steps with the same parent? Or does this not matter for your use case?

**Comment by kronok** [comment]
Ah, I felt like it really wanted a position field on my model, but I was looking at the CTENode class thinking if it wanted it, it would be in the CTENode class. Figured some magic was happening somewhere that I wasn't seeing.

I don't really need to order steps with the same parent, or haven't figured out that I need that yet, but I'm cool with having that ability.

I actually don't do the whole CMS thing (so I don't use the Page class at all), I just mainly use content-blocks and dabbling with cte-forest (previously mptt), but I like the better html renderer and plugins that feincms3 supplies at the moment.

Anyways, I added a position field and it's working. Thanks for showing me that line!
